conn_http.go

```diff
@@ -69,15 +89,47 @@ type httpConnect struct {
 	location  *time.Location
 }
 
-func (h *httpConnect) prepareRequest(ctx context.Context, query string, args ...interface{}) (*http.Request, error) {
-	query, err := bind(h.location, query, args)
+func writeData(encoder *binary.Encoder, block *proto.Block, name string) error {
+	return block.Encode(encoder, 0)
+}
+
+func readData(decoder *binary.Decoder) (*proto.Block, error) {
+	var block proto.Block
+	if err := block.Decode(decoder, 0); err != nil {
+		return nil, err
+	}
+	return &block, nil
+}
+
+func readResponse(response *http.Response) ([]byte, error) {
+	var result []byte
+	if response.ContentLength > 0 {
+		result = make([]byte, 0, response.ContentLength)
+	}
+	buf := bytes.NewBuffer(result)
+	defer response.Body.Close()
+	_, err := buf.ReadFrom(response.Body)
+
 	if err != nil {
 		return nil, err
 	}
 
-	reader := strings.NewReader(query)
+	return buf.Bytes(), nil
+}
+
+func (h *httpConnect) prepareRequest(ctx context.Context, reader io.Reader, extra map[string]string) (*http.Request, error) {
+
+	u := *h.url
+
+	if len(extra) != 0 {
+		query := u.Query()
+		for key, value := range extra {
+			query.Set(key, value)
+		}
+		u.RawQuery = query.Encode()
+	}
 
-	req, err := http.NewRequest(http.MethodPost, h.url.String(), reader)
+	req, err := http.NewRequest(http.MethodPost, u.String(), reader)
```

Comment:
Can we use the context here? i.e. NewRequestFromContext - thus allowing time outs etc to be propagated?

Reply:
Added